Upgrade react-native-video and drop jetifier on Android

mchowning opened the pull request:
⚠️  Need to update the react-native-dependency to reference a `.tgz` file before merging this.  ⚠️

Related gutenberg-mobile PR: https://github.com/wordpress-mobile/react-native-video/commit/5c6697afc5244883615acc84f054a1ed2a0c6ed0

## Description
This PR updates react-native-video to allow us to drop jetifier on Android.

## Testing Instructions
Using the demo app, play the video from the sample content. On Android this should route to an external application (probably the web browser) and either play the video or offer to download it. On iOS though, the demo video should open and play within the demo app itself.

## Screenshots <!-- if applicable -->

## Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->

## Checklist:
- [X] My code is tested.
- [X] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [X] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [ ] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [X] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [X] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [X] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/code/native-mobile.md -->
- [X] I've updated related schemas if appropriate. <!-- Reference: https://github.com/WordPress/gutenberg/tree/trunk/schemas -->

mchowning committed:
Disable jetifier

mchowning commented:
I'm going to keep this a draft until we update the react-native-video dependency to not reference a git commit hash, which we can do once https://github.com/wordpress-mobile/react-native-video/pull/15 is merged).

mchowning committed:
Update react-native-video

hypest commented on package-lock.json:
```diff
@@ -49712,10 +49762,12 @@
-			"version": "https://raw.githubusercontent.com/wordpress-mobile/react-native-video/5.0.2-wp-2/react-native-video-5.0.2-wp-2.tgz",
-			"integrity": "sha512-JcZKEFDbje8MNEd1sEO5eQDrF7/BUaYMskI2G7rTXd0EwOPGx9rHOdx0uAmmew+NvkMNPmh+H6uKsgiaS891dA==",
+			"version": "git+https://github.com/wordpress-mobile/react-native-video.git#31ebacf5584219d4b3888b95673c35243fd30d2e",
+			"from": "git+https://github.com/wordpress-mobile/react-native-video.git#31ebacf5584219d4b3888b95673c35243fd30d2e",
```
Not sure, will there be a problem that the `integrity` field is now missing? 🤔

mchowning replied:
I believe that field will come back when I switch from a github commit to a tar.gz file  since [it sounds like](https://docs.npmjs.com/cli/v6/configuring-npm/package-lock-json#integrity) when using a git commit just the commit hash itself is used to insure integrity. 

With that said, the static lint checks are failing because of the package-lock file, and I haven't been able to figure out what is going on with that yet.

mchowning committed:
Fresh package-lock file

mchowning committed:
Update package-lock.json file

1. Copied package-lock.json file from trunk
2. Removed react-native-video dep from
   packages/react-native-editor/package.json
3. Ran `npm install`
4. Restored react-native-video dep
5. Ran `npm install`

mchowning committed:
react-native-video version without apache deps

mchowning committed:
Removed react-native-video

mchowning committed:
Add react-native-video back

mchowning committed:
Reset package-lock.json

mchowning committed:
Partial package-lock.json update

mchowning committed:
Reset package-lock.json

mchowning committed:
Add react-native-video back in

mchowning commented:
Marking this as ready for review, but we will still need to update the react-native-video ref to point to the tag and not the branch once https://github.com/wordpress-mobile/react-native-video/pull/15 is merged and we create the `5.2.0-wp-1` tag.

hypest reviewed (approved):
LGTM!

mchowning committed:
Update react-native-video dependency

mchowning committed:
Update package-lock.json

mchowning committed:
Update react-native-video to reference tag version

mchowning committed:
Make usage of tabs consistent
